packages/api-plugin-simple-schema/src/util/dbVersionCheck.js

```diff
@@ -0,0 +1,54 @@
+import doesDatabaseVersionMatch from "@reactioncommerce/db-version-check";
+import { migrationsNamespace } from "../../migrations/migrationsNamespace.js";
+
+const expectedVersion = 2;
+
+const simpleSchemaIntrospectPermissions = [
+  "reaction:legacy:simpleSchema/introspect:*",
+  "reaction:legacy:simpleSchema/introspect:Cart",
+  "reaction:legacy:simpleSchema/introspect:Product",
+  "reaction:legacy:simpleSchema/introspect:Order",
+  "reaction:legacy:simpleSchema/introspect:Account",
+  "reaction:legacy:simpleSchema/introspect:Promotion"
+];
+
+/**
+ * @summary Checks if the version of the database matches requirement
+ * @param {Object} context Startup context
+ * @returns {undefined}
+ */
+export default async function dbVersionCheck(context) {
+  const setToExpectedIfMissing = async () => {
+    // Check if atleast one of the simpleSchema introspect permissions exist in the roles collection
+    const allRoles = await context.collections.roles.find({}).toArray();
+    let rolesFound = false;
+    if (allRoles && Array.isArray(allRoles) && allRoles.length) {
+      rolesFound = allRoles.some((role) => simpleSchemaIntrospectPermissions.includes(role.name));
+    }
```

Comment:
The above could be simplified and optimized to

```js
const existingRole = await context.collections.roles.findOne({ name: { $in: simpleSchemaIntrospectPermissions } });
const rolesFound = !!existingRole;
```

Reply:
Updated as recommended